scripts/autorelease.sh

```diff
@@ -27,21 +29,36 @@ shopt -s extglob
 cp -L "$TEMPDIR"/"$project"/!(*.md) "$assets_dir"
 # Iterate over assets, calculate sha256sum and sign them in order
 # to include this additional info to the release assets as well
-for asset in $assets_dir/*; do
+for asset in "$assets_dir"/*; do
     sha256sum "$asset" | sed 's/ .*/ /' > "$asset.sha256"
     gpg --armor --detach-sign "$asset"
 done
 
-# Delete release if it exists
-gh release delete auto-release --yes || true
+msg_regex="Merge pull request .* from serokell\/auto\/(v.*)-release"
+rc_regex="^v[0-9]+\.[0-9]+-rc[0-9]+"
+
+# We create a pre-release in any case but if we are merging a stable version release PR
+mode_flag="--prerelease"
+
+if [[ ${1-""} =~ $msg_regex ]]; then
+    tag="${BASH_REMATCH[1]}-$(jq -r '.release' meta.json)"
+    if [[ ! $tag =~ $rc_regex ]]; then
+        mode_flag=""
+    fi
+else
+    tag="auto-release"
+
+    # Delete autorelease if it exists
+    gh release delete auto-release --yes || true
+fi
 
 # Update the tag
 git fetch # So that the script can be run from an arbitrary checkout
-git tag -f auto-release
+git tag -f $tag
 git push --force --tags
 
 # Create release
-gh release create -F "$TEMPDIR"/"$project"/release-notes.md --prerelease auto-release --title auto-release
+gh release create -F "$TEMPDIR"/"$project"/release-notes.md "$mode_flag" $tag --title $tag
```

Comment:
I'm slightly afraid that here `gh` might not like `"$mode_flag"` here when the variable is empty because it will use it as an empty argument :thinking:
E.g.
```
$ dir=""
$ git log "$dir"
fatal: empty string is not a valid pathspec. please use . instead if you meant to match all paths
```

Reply:
🤔 You're right, might be a problem. As `$mode_flag` can only ever take one of two values (`""` or `"--prerelease"), I think it's fine to leave out the quotes to let it expand to nothing

Reply:
I mean `$mode_flag` expanding to nothing may cause a falure

Reply:
Because no argument and nothing argument may be handle differently, see my example with `git log`

Reply:
oh, I see now that you've changed